src/Telegram.Bot/Requests/SetWebhookRequest.cs

```diff
@@ -0,0 +1,115 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using Newtonsoft.Json;
+using Telegram.Bot.Helpers;
+using Telegram.Bot.Types.Enums;
+
+namespace Telegram.Bot.Requests
+{
+    /// <summary>
+    /// Specify a url and receive incoming updates via an outgoing webhook
+    /// </summary>
+    public class SetWebhookRequest : RequestBase<bool>
+    {
+        private IEnumerable<UpdateType> _allowedUpdateTypes;
+
+        /// <summary>
+        /// HTTPS url to send updates to. Use an empty string to remove webhook integration.
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public string Url { get; set; }
+
+        /// <summary>
+        /// Public key certificate so that the root certificate in use can be checked
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public Stream Certificate { get; set; }
+
+        /// <summary>
+        /// Maximum allowed number of simultaneous HTTPS connections to the webhook for update delivery
+        /// </summary>
+        [JsonProperty(DefaultValueHandling = DefaultValueHandling.Ignore)]
+        public int MaxConnections { get; set; }
```

Comment:
Values are _1-100. Defaults to 40_. We could statically enforce the range better by using `uint` instead of `int`.

Reply:
Oh, I forgot about this constraint. It can be implemented in the getter. So all values less than 1 or greater than 100 would not be sent. Though we can't enforce this in `ITelegramBotClient.SetWebhookAsync` and it would be possible to pass any valid number to it.

Reply:
Request validation is another nice feature we can implement. We'll look into it later. For now, just a simple type change would help a bit.